xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h

```diff
@@ -1266,6 +1268,44 @@ namespace vfs
     //==========================================================================
     ///
     /// @ingroup cpp_kodi_vfs_CFile
+    /// @brief retrieve a response header
+    ///
+    /// @param[in] name         The name of the response header to retrieve the value for
+    /// @return                 value of requested response header, empty on failure
+    ///
+    std::string CURLGetResponseHeader(const std::string &name)
+    {
+      if (!m_file)
+      {
+        kodi::Log(ADDON_LOG_ERROR, "kodi::vfs::CURLCreate(...) needed to call before CURLGetResponseHeader!");
+        return "";
+      }
+      return ::kodi::addon::CAddonBase::m_interface->toKodi->kodi_filesystem->curl_get_response_header(::kodi::addon::CAddonBase::m_interface->toKodi->kodiBase, m_file, name.c_str());
+    }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// @ingroup cpp_kodi_vfs_CFile
+    /// @brief retrieve the response status code
+    ///
+    /// @return      response status code (e.g. 200 for HTTP requests)
+    ///
+    int CURLGetResponseStatusCode()
+    {
+      if (!m_file)
+      {
+        kodi::Log(ADDON_LOG_ERROR, "kodi::vfs::CURLCreate(...) needed to call before CURLGetResponseStatusCode!");
+        return -1;
```

Comment:
Magic numbers all over the place... As this is official API, we should be very clean here..